### PR TITLE
vo_dmabuf_wayland: improve resize and buffer pool reset

### PR DESCRIPTION
**Pull request by boxerab**
This PR makes a few small improvements to how this vo handles resize and reset of pool.

### CODE REVIEW

**Comment by t-8ch** [comment]
This doesn't seem to work.

This works for me:
```diff
diff --git a/video/out/vo_dmabuf_wayland.c b/video/out/vo_dmabuf_wayland.c
index 565fe50a97d9..0526fa0646ca 100644
--- a/video/out/vo_dmabuf_wayland.c
+++ b/video/out/vo_dmabuf_wayland.c
@@ -208,6 +208,8 @@ static void resize(struct vo *vo)
 
     vo->want_redraw = true;
     p->resized = true;
+    p->want_reset = true;
+    p->reset_count = 0;
 }
 
 static void draw_frame(struct vo *vo, struct vo_frame *frame)
```

**Comment by boxerab** [comment]
@t-8ch thanks, please can you move your comment to the original issue ? We can discuss there

**Comment by Dudemanguy** [comment]
Having to add the purgatory list makes this way more complicated imo. Surely there's a better way to fix the issue.

**Review by Dudemanguy — APPROVED** [review]
Seems to work and solves a real issue.

File: video/out/vo_dmabuf_wayland.c [hunk 1]
```diff
@@ -244,6 +236,13 @@ static void draw_frame(struct vo *vo, struct vo_frame *frame)
...
         return;
 
+    // ensure the pool is reset after hwdec seek,
+    // to avoid stutter artifact
+    if (p->want_reset)
+        wlbuf_pool_clean(p->wlbuf_pool,false);
+    if (p->want_resize)
+        resize(vo);
+
     MP_VERBOSE(entry->vo, "Schedule buffer pool entry : %lu\n",entry->key );
```
Comment (Dudemanguy) [inline video/out/vo_dmabuf_wayland.c:246]: Woops missed this. Could you make this `MP_TRACE` as well since it's per frame?

Comment (boxerab) [reply video/out/vo_dmabuf_wayland.c]: thx, fixed.

**Pull request merged**